app/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source.rb

```diff
@@ -33,4 +36,49 @@ def sync
     update!(:status => "error", :last_updated_on => Time.zone.now, :last_update_error => error)
     raise error
   end
+
+  private
+
+  # Find Terraform Templates(dir) in the git repo.
+  # Iterate throught git repo worktree, and collate all terraform template dir's (dirs with .tf or .tf.json files).
+  #
+  # Returns [hashmap] of template dirs and files within it.
+  def find_templates_in_git_repo()
+    template_dirs = {}
+
+    # checkout files to temp dir, we need for prasing input/output vars
+    git_checkout_tempdir = Dir.mktmpdir("terraform-git")
+    checkout_git_repository(git_checkout_tempdir)
+
+    # traverse through files in git-worktree
+    git_repository.with_worktree do |worktree|
+      worktree.ref = scm_branch
+
+      # Find all dir's with .tf/.tf.json files
+      worktree.blob_list.each do |filepath|
+        next unless filepath.end_with?(".tf", ".tf.json")
+
+        parent_dir = File.dirname(filepath)
+        next if template_dirs.has_key?(parent_dir)
```

Comment:
```suggestion
        next if template_dirs.key?(parent_dir)
```

Minor suggestion above.